### PR TITLE
[fix](hist) Fix unstable result of aggregrate function hist

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
* Target

Fix unstable result of hist function when involving null value.

* Reproduce

test result of `regression-test/suites/query_p0/sql_functions/aggregate_functions/test_aggregate_all_functions2.groovy` is unstable, sql `SELECT histogram(k7, 5) FROM baseall` will sometimes acts like the second argument is not passed in.

* Root reason

We have short-circuit in AggregateFunctionNullVariadicInline, when this row is NULL, the value will not be added by the nested function. Implementation of histogram relies on its add method to get its seconds argument, when we have an all null value block, histogram will not get its seconds arg even if sql is like `select(k7, 5)`, so a max_bucket_num with default value 128 is serialized. When we do merging, and happens to deserialize the above block at last, the max_bucket_num in merge stage will be assigned to 128, and this leads to the wrong result.

* Fix by

Init value of max_bucket_num is assigned to 0, when we do merging, we will discard this aggregated data if its max_bucket_num is 0.

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/vec/aggregate_functions/aggregate_function_histogram.h [hunk 1]
```diff
@@ -186,13 +198,18 @@ class AggregateFunctionHistogram final
...
+        if constexpr (has_input_param) {
+            Int32 input_max_num_buckets =
+                    assert_cast<const ColumnInt32*>(columns[1])->get_element(row_num);
+            if (input_max_num_buckets <= 0) {
+                throw doris::Exception(ErrorCode::INVALID_ARGUMENT,
+                                       "Invalid max_num_buckets {}, row_num {}",
+                                       input_max_num_buckets, row_num);
+            }
             this->data(place).set_parameters(
                     assert_cast<const ColumnInt32*>(columns[1])->get_element(row_num));
```
Comment (zhangstar333) [inline be/src/vec/aggregate_functions/aggregate_function_histogram.h:210]: here could use input_max_num_buckets directly

File: be/src/vec/aggregate_functions/aggregate_function_histogram.h [hunk 2]
```diff
@@ -86,7 +92,9 @@ struct AggregateFunctionHistogramData {
     }
 
     void merge(const AggregateFunctionHistogramData& rhs) {
-        if (!rhs.max_num_buckets) {
+        // if rhs.max_num_buckets == 1, it means the input block for serialization is all null
```
Comment (zhangstar333) [inline be/src/vec/aggregate_functions/aggregate_function_histogram.h:95]: rhs.max_num_buckets == 1,  seems is means == -1?

Comment (zhiqiang-hhhh) [reply be/src/vec/aggregate_functions/aggregate_function_histogram.h:95]: yes, i will fix

File: be/src/vec/aggregate_functions/aggregate_function_histogram.h (hunk 1 above)
Comment (zhiqiang-hhhh) [reply be/src/vec/aggregate_functions/aggregate_function_histogram.h:210]: yes

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
@zhangstar333 I changed the default value from -1 to 0, so that we can fix the problem in an upgrade friendly way.

**Comment by zhiqiang-hhhh** [comment]
run external

**Review by zhangstar333 — APPROVED** [review]

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**